src/Humanizer/Localisation/NumberToWords/SpanishNumberToWordsConverter.cs

```diff
@@ -1,193 +1,383 @@
+                return "cero";
+            }
 
-                number %= 1000000000;
+            if (number < 0)
+            {
+                return ConvertToOrdinal(Math.Abs(number), gender);
             }
 
-            if ((number / 1000000) > 0)
+            if (IsRoundBillion(number))
             {
-                parts.Add(number / 1000000 == 1
-                    ? "un millón"
-                    : string.Format("{0} millones", Convert(number / 1000000)));
+                return ConvertRoundBillionths(number, gender);
+            }
 
-                number %= 1000000;
+            if (IsRoundMillion(number))
+            {
+                return ConvertToOrdinal(number / 1000, gender).Replace("milésim", "millonésim");
             }
 
-            if ((number / 1000) > 0)
+            wordBuilder.Add(ConvertTensAndHunderdsOfThousandths(number, out var remainder, gender));
+            wordBuilder.Add(ConvertThousandths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertHundredths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertTenths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertOrdinalUnits(remainder, gender, wordForm));
+
+            return BuildWord(wordBuilder);
+        }
+
+        public override string ConvertToTuple(int number)
+        {
+            string[] map = {
+                "cero veces", "una vez", "doble", "triple", "cuádruple", "quíntuble", "séxtuple", "séptuple", "óctuple",
+                "nonupla", "décuplo", "undécuplo", "duodécuplo", "terciodécuplpo" };
+
+            number = Math.Abs(number);
+
+            if (number < map.Length)
+                return map[number];
+
+            return Convert(number) + " veces";
+        }
+
+        private static string BuildWord(IReadOnlyList<string> wordParts)
+        {
+            var parts = wordParts.ToList();
+            parts.RemoveAll(string.IsNullOrEmpty);
+            return string.Join(" ", parts);
+        }
+
+        private static string ConvertHundreds(in long inputNumber, out long remainder, GrammaticalGender gender)
+        {
+            var wordPart = string.Empty;
+            remainder = inputNumber;
+
+            if ((inputNumber / 100) > 0)
             {
-                parts.Add(number / 1000 == 1
-                    ? "mil"
-                    : string.Format("{0} mil", Convert(number / 1000, gender)));
+                wordPart = inputNumber == 100 ?
+                    "cien" :
+                    GetGenderedHundredsMap(gender)[(int)(inputNumber / 100)];
 
-                number %= 1000;
+                remainder = inputNumber % 100;
             }
 
-            if ((number / 100) > 0)
+            return wordPart;
+        }
+
+        private static string ConvertHundredths(in int number, out int remainder, GrammaticalGender gender)
+        {
+            string[] hundredthsRootMap = {
```

Comment:
Same as above, should be a static member on the type

Reply:
No problem